### PR TITLE
Keras-mxnet 2.2.4 release

### PR DESCRIPTION
**Pull request by roywei**
### Summary

- Pull all changes from keras-team/keras 2.2.4 (commit d059890d0342955e968fdf97b5a90d19c9d68b4e)
- fixed conflicts
- resolved failing unit tests and style checks

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n]

### CODE REVIEW

**Comment by kalyc** [comment]
Can you add link to the pickle issue that is opened with keras-team for mxnet backend?

**Comment by roywei** [comment]
@kalyc it been tracked here:https://github.com/awslabs/keras-apache-mxnet/issues/194

**Comment by kalyc** [comment]
Related: https://github.com/awslabs/keras-apache-mxnet/issues/194

File: setup.py [hunk 1]
```diff
@@ -36,16 +36,17 @@
       install_requires=['numpy>=1.9.1',
                         'scipy>=0.14',
                         'six>=1.9.0',
-                        'h5py>=2.7.1',
                         'pyyaml',
-                        'keras_applications==1.0.4',
-                        'keras_preprocessing==1.0.1'],
+                        'h5py',
```
Comment (kalyc) [inline setup.py]: is this compatible with older versions of h5py? we should probably keep the version requirement of h5py to prevent any breaking builds

Comment (roywei) [reply setup.py]: this change is from keras, let's keep it and it's not causing any build breaks: https://github.com/keras-team/keras/blob/master/setup.py

**Pull request merged**